ArduPlane/px4_mixer.cpp

```diff
@@ -42,173 +43,222 @@ bool Plane::print_buffer(char *&buf, uint16_t &buf_size, const char *fmt, ...)
     return true;
 }
 
+/* 
+   create a mixer for a normal angle channel
+*/
+bool Plane::mix_one_channel(char *&buf, uint16_t &buf_size, uint8_t out_chan, uint8_t in_chan)
+{
+    const float limit = 10000;
+    const RC_Channel *inch = RC_Channels::rc_channel(in_chan);
+    const SRV_Channel *outch = SRV_Channels::srv_channel(out_chan);
+    const float scale_out = limit*(outch->get_output_max() - outch->get_output_min()) / (PX4_LIM_RC_MAX - PX4_LIM_RC_MIN);
+
+    bool is_throttle = in_chan==rcmap.throttle()-1;
+    int16_t outch_trim = is_throttle?1500:outch->get_trim();
+
+    outch_trim = constrain_int16(outch_trim, outch->get_output_min()+1, outch->get_output_max()-1);
+
+    if (!print_buffer(buf, buf_size, "M: 1\n")) {
+        return false;
+    }
+
+    int32_t out_min = scale_out*(1500 - outch->get_output_min()) / (1500 - PX4_LIM_RC_MIN);
+    int32_t out_max = scale_out*(outch->get_output_max() - 1500) / (PX4_LIM_RC_MAX - 1500);
+    int32_t out_trim = scale_out*(outch_trim - 1500) / ((PX4_LIM_RC_MAX - PX4_LIM_RC_MIN) / 2);
+    int32_t in_mul = inch->get_reverse() == outch->get_reversed()?1:-1;
+
+    // note that RC trim is taken care of in rc_config
+    float scale_low =  limit*(1500 - PX4_LIM_RC_MIN)/(1500 - inch->get_radio_min());
+    float scale_high = limit*(PX4_LIM_RC_MAX - 1500)/(inch->get_radio_max() - 1500);
+    
+    if (!print_buffer(buf, buf_size, "O: %d %d %d %d %d\n",
+                      int(out_min),
+                      int(out_max),
+                      int(out_trim),
+                      int(-limit), int(limit))) {
+        return false;
+    }
+    if (!print_buffer(buf, buf_size, "S: 0 %u %d %d %d %d %d\n",
+                      in_chan,
+                      int(in_mul*scale_low), int(in_mul*scale_high),
+                      0,
+                      int(-limit), int(limit))) {
+        return false;
+    }
+    return true;
+}
+
+/*
+  mix two channels using elevon style mixer
+*/
+bool Plane::mix_two_channels(char *&buf, uint16_t &buf_size, uint8_t out_chan, uint8_t in_chan1, uint8_t in_chan2, bool left_channel)
+{
+    const float limit = 10000;
+    const RC_Channel *inch1 = RC_Channels::rc_channel(in_chan1);
+    const RC_Channel *inch2 = RC_Channels::rc_channel(in_chan2);
+    const SRV_Channel *outch = SRV_Channels::srv_channel(out_chan);
+    const float scale_out = limit*(outch->get_output_max() - outch->get_output_min()) / (PX4_LIM_RC_MAX - PX4_LIM_RC_MIN);
+    int16_t outch_trim = outch->get_trim();
+
+    outch_trim = constrain_int16(outch_trim, outch->get_output_min()+1, outch->get_output_max()-1);
+
+    if (!print_buffer(buf, buf_size, "M: 2\n")) {
+        return false;
+    }
+
+    int32_t out_min = scale_out*(1500 - outch->get_output_min()) / (1500 - PX4_LIM_RC_MIN);
```

Comment:
My main question here is why isn't this using the output channel trim rather then 1500? I've setup planes where 1500 is outside the min/max range which means that the number here would be wildly off from what it's expected to be... (This concern repeats itself a number of times below here)

I just haven't gotten the time to grab you and talk through this in more detail.